include/openthread/thread.h

```diff
@@ -663,6 +665,237 @@ otError otThreadGetParentAverageRssi(otInstance *aInstance, int8_t *aParentRssi)
  */
 otError otThreadGetParentLastRssi(otInstance *aInstance, int8_t *aLastRssi);
 
+/**
+ * Maximum Number of Network Diagnostic TLV Types to Request or Reset.
+ */
+#define OT_NETWORK_DIAGNOSTIC_TYPELIST_MAX_ENTRIES 19
```

Comment:
how about we move all network diagnostic related definitions into a separate file, such as `openthread/netdiag.h`?

Reply:
done.